### ISSUE TITLE
403 error with segments in multi site setup

### ISSUE THREAD

**Issue by joostclevis** [putyourlightson/craft-campaign#504]
### Bug Report

We have a multi site setup. Both sites use Campaign. When creating a segment on the first site it works fine. But when adding an segment on the second site and then opening that segment results in a error:

```php
2024-11-22 10:16:35 [-][1813][i6rbbbdjiprs6dvlp4hbrcg67k][error][yii\web\HttpException:403] yii\web\ForbiddenHttpException: User not permitted to access this source in /projects/rnlproduc_c8/releases/bd72b440185ea1f9947cf0fa82f3ff1979babb84/vendor/craftcms/cms/src/controllers/ElementIndexesController.php:431
Stack trace:
#0 /projects/rnlproduc_c8/releases/bd72b440185ea1f9947cf0fa82f3ff1979babb84/vendor/craftcms/cms/src/controllers/ElementIndexesController.php(99): craft\controllers\ElementIndexesController->source()
#1 /projects/rnlproduc_c8/releases/bd72b440185ea1f9947cf0fa82f3ff1979babb84/vendor/yiisoft/yii2/base/Controller.php(176): craft\controllers\ElementIndexesController->beforeAction()
#2 /projects/rnlproduc_c8/releases/bd72b440185ea1f9947cf0fa82f3ff1979babb84/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction()
#3 /projects/rnlproduc_c8/releases/bd72b440185ea1f9947cf0fa82f3ff1979babb84/vendor/craftcms/cms/src/web/Application.php(295): yii\base\Module->runAction()
#4 /projects/rnlproduc_c8/releases/bd72b440185ea1f9947cf0fa82f3ff1979babb84/vendor/craftcms/cms/src/web/Application.php(608): craft\web\Application->runAction()
#5 /projects/rnlproduc_c8/releases/bd72b440185ea1f9947cf0fa82f3ff1979babb84/vendor/craftcms/cms/src/web/Application.php(274): craft\web\Application->_processActionRequest()
#6 /projects/rnlproduc_c8/releases/bd72b440185ea1f9947cf0fa82f3ff1979babb84/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest()
#7 /projects/rnlproduc_c8/releases/bd72b440185ea1f9947cf0fa82f3ff1979babb84/public_html/index.php(21): yii\base\Application->run()
#8 {main}
``` 

### Plugin Version

1.26.0

### Craft CMS Version

 3.9.13 

### PHP Version

8.1.30

**Comment by bencroker** [comment]
I just tested locally using Campaign 1.26.0 and Craft 3.9.13 and it works as expected. Are you logged in as an admin user? The error message `User not permitted to access this source` suggests that this might be a permissions issue.

**Comment by joostclevis** [comment]
@bencroker this also happens if logged in as admin user.

**Comment by bencroker** [comment]
I’m not sure where to begin troubleshooting this as I can’t replicate locally and, as you can see from the stack trace, no Campaign files are involved in the exception, only Craft and Yii files.

**Comment by joostclevis** [comment]
@bencroker i get that, does it help if we give you access to the site?

**Comment by bencroker** [comment]
Yes, I can take a look. Please send credentials to support@putyourlightson.com.

**Comment by bencroker** [comment]
Fixed in https://github.com/putyourlightson/craft-campaign/commit/6668aadc43b314cc442192e994e80fe6ac03c2af and released in [1.26.1](https://github.com/putyourlightson/craft-campaign/releases/tag/1.26.1).
